src/portable/st/synopsys/dcd_synopsys.c

```diff
@@ -92,6 +93,18 @@
 #define EP_MAX_FS       6
 #define EP_FIFO_SIZE_FS 1280
 
+#elif CFG_TUSB_MCU == OPT_MCU_GD32VF103
+#define STM32F1_SYNOPSYS
+#include "gd32vf103.h"
+#include "nmsis_core.h"
```

Comment:
I am not familiar with riscv and I am not sure if nuclei nmsis is as universal used as arm cmsis. If we didn't use it much e.g only enable/disable IRQ, maybe it is best to just use the low level code from giga firmware. That will allow user to run tinyusb with bare metal without the need for nuclei sdk.

Reply:
The whole risc-v ecosystem is really young and because of the decentralized and open nature of the ISA I doubt that something universal as CMSIS is going to emerge. So NMSIS is largely Nuclei specific as the processor core have vendor specific extensions but very well designed in my opinion. It also provides a CMSIS compatibility layer. Although for this port very few parts are actually used.

Reply:
looking more at nmsis, I think it is a great repo. I will leave to you to decide whether we should use nmsis here. Note that, anything used in this dcd file will be considered as tinyusb dependency.

Reply:
That sounds good! See my other comment for more rationale.

Reply:
I actually take this back, given the fact that nmsis requires firmware lib modification and couldn't work with official lib downloaded from Giga. Since it is simple enough, probably only ECLIC_EnableIRQ/DisableIRQ(), do you think we could do it with register level, I actually have to go this low for some of mcu ports to stay independent enough.

Reply:
This is absolutely possible! Enabling and Disabeling the IRQs is a simple setting and clearing the MIE bit in the mstatus register. I'll figure out if we go with the gd32vf103 firmware library provided functions or some inline asm.

Reply:
Oh except for actually enabling and disabling a specific IRQ, but this is rather easy to do.

Reply:
Thank you, we only need to specifically enable/disable the USB IRQ only, and don't need a generic function like NVIC. And bit set/clear with correct mask and register would do the trick hopefully

Reply:
Please have a look at https://github.com/hathach/tinyusb/pull/959/commits/c6d495d6436c797b4efd029bd16ee8f19ff0a9f0 this is the commit that removes any external dependency for the core code. Are you fine with helper functions being directly in the driver code?